framework/src/utils/MooseUtils.C

```diff
@@ -561,6 +564,46 @@ toUpper(const std::string & name)
   return upper;
 }
 
+ExecFlagEnum
+getDefaultExecFlagEnum()
+{
+  ExecFlagEnum exec_enum = ExecFlagEnum();
+  exec_enum.addAvailableFlags(EXEC_NONE,
+                              EXEC_INITIAL,
+                              EXEC_LINEAR,
+                              EXEC_NONLINEAR,
+                              EXEC_TIMESTEP_END,
+                              EXEC_TIMESTEP_BEGIN,
+                              EXEC_FINAL,
+                              EXEC_CUSTOM);
+  return exec_enum;
+}
+
+template <typename T>
+T
+stringToNumber(const std::string & input)
+{
+  T output;
+  std::istringstream ss(input);
+  if (!(ss >> output) || !ss.eof())
+    mooseError(
+        "Failed to convert '", input, "' to the supplied type of ", typeid(output).name(), ".");
+  return output;
+}
+
+// Explict instantiation of fundamental numeric types
+template short int stringToNumber<short int>(const std::string & input);
+template unsigned short int stringToNumber<unsigned short int>(const std::string & input);
+template int stringToNumber<int>(const std::string & input);
+template unsigned int stringToNumber<unsigned int>(const std::string & input);
+template long int stringToNumber<long int>(const std::string & input);
+template unsigned long int stringToNumber<unsigned long int>(const std::string & input);
+template long long int stringToNumber<long long int>(const std::string & input);
+template unsigned long long int stringToNumber<unsigned long long int>(const std::string & input);
+template float stringToNumber<float>(const std::string & input);
+template double stringToNumber<double>(const std::string & input);
+template long double stringToNumber<long double>(const std::string & input);
```

Comment:
I wouldn't bother with this. This utility isn't used too often. Let's just move the template to the header file and remove these.

Reply:
I disagree, I don't like to create a template function that can be called with types that don't work. But, if you don't want them I will pull them out.

Reply:
That's not what happens with template functions. You only need explicit instantiation if you put the body in the .C file. If it's in the header file, all necessary instantiations are generated on-demand so it works with every type.

Reply:
I agree, but it doesn't work with every type. What if I give is a MooseObject? I know that it is stupid, but you can do it.

Reply:
Still works as long as the definition is in the header file.